spec/requests/harvest/harvest_spec.rb

```diff
@@ -293,6 +293,61 @@
 
       expect_error(:unprocessable_content, /found unpermitted parameter: :banana/)
     end
+
+    context 'when sites are deleted' do
+      let!(:site1) { Creation::Common.create_site(owner_user, project, region:, name: 'meeseeks') }
+
+      before do
+        create_harvest(streaming: false)
+        expect(harvest.mappings.length).to eq 2
+
+        site1.delete
+      end
+
+      it 'cleans up mappings referencing deleted sites when saved' do
+        harvest.save
```

Comment:
Maybe .save!

Also add a comment like

    # before mappings were cleaned up we would fail validation if a site was missing